schemas/defs.schema.json

```diff
@@ -143,6 +172,74 @@
         "status"
       ],
       "type": "object"
+    },
+    "GroupData": {
+      "additionalProperties": false,
+      "properties": {
+        "name": {
+          "description": "Short name",
+          "type": "string"
+        },
+        "parent": {
+          "description": "Identifier of parent group",
+          "type": "string"
+        }
+      },
+      "required": [
+        "name"
+      ],
+      "type": "object"
+    },
+    "SnapshotData": {
+      "additionalProperties": false,
+      "properties": {
+        "name": {
+          "description": "Short name",
+          "type": "string"
+        },
+        "spec": {
+          "description": "Specification",
+          "format": "uri",
+          "type": "string"
+        }
+      },
+      "required": [
+        "name",
+        "spec"
+      ],
+      "type": "object"
+    },
+    "WebFeaturesData": {
+      "additionalProperties": false,
+      "properties": {
+        "features": {
+          "additionalProperties": {
+            "$ref": "#/definitions/FeatureData"
+          },
+          "description": "Feature identifiers and data",
+          "type": "object"
```

Comment:
The thing you'll lose here is the `patternProperties` constraint from `features.schema.json`:

```
  "patternProperties": {
    "^[a-z0-9-]*$": {
      "$ref": "defs#/definitions/FeatureData"
    }
  }
```

That said, if I rename a file to use underscores, `ajv` seems to not catch the break with the schema. So maybe this doesn't matter, or never has? 🤷

See https://github.com/web-platform-dx/web-features/pull/44#issuecomment-1438664927 for background

Reply:
I tried to find a way to express this with TypeScript but failed. It may be possible with https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html but it wouldn't be pretty, and I can't imagine it would get turned into a pretty RegEx in JSON schema.

Maybe https://github.com/web-platform-dx/web-features/pull/1317 instead?